src/features/events/components/EventContactCard.tsx

```diff
@@ -0,0 +1,157 @@
+import { Box, Button, Typography } from '@mui/material';
+import {
+  Close,
+  FaceOutlined,
+  FaceRetouchingOffOutlined,
+} from '@mui/icons-material';
+import { FC, useContext } from 'react';
+
+import EventDataModel from 'features/events/models/EventDataModel';
+import messageIds from 'features/events/l10n/messageIds';
+import { useMessages } from 'core/i18n';
+import { ZetkinEvent } from 'utils/types/zetkin';
+import ZUIAvatar from 'zui/ZUIAvatar';
+import ZUICard from 'zui/ZUICard';
+import { ZUIConfirmDialogContext } from 'zui/ZUIConfirmDialogProvider';
+import { MUIOnlyPersonSelect as ZUIPersonSelect } from 'zui/ZUIPersonSelect';
+
+interface EventContactCardProps {
+  data: ZetkinEvent;
+  model: EventDataModel;
+  orgId: number;
+}
+
+interface ContactDetailsProps {
+  contact: { id: number; name: string };
+  model: EventDataModel;
+  orgId: number;
+}
+
+interface ContactSelectProps {
+  model: EventDataModel;
+}
+
+const ContactDetails: FC<ContactDetailsProps> = ({ contact, model, orgId }) => {
+  const messages = useMessages(messageIds);
+  const { showConfirmDialog } = useContext(ZUIConfirmDialogContext);
+
+  return (
+    <>
+      <Box m={1} sx={{ display: 'inline-block', verticalAlign: 'middle' }}>
+        <ZUIAvatar orgId={orgId} personId={contact.id} />
+      </Box>
+      <Typography sx={{ display: 'inline-block', verticalAlign: 'middle' }}>
+        {contact.name}
+      </Typography>
+      <Box m={1}>
+        <Button
+          onClick={() => {
+            showConfirmDialog({
+              onSubmit: () => {
+                model.removeContact();
+              },
+              warningText: messages.eventContactCard.warningText({
+                name: contact!.name,
```

Comment:
The "non-null assertion" operator (`!`) should be avoided, and I think it's not even needed in this case (because the type of `contact` does not allow `null`).